src/Admin/AbstractAdmin.php

```diff
@@ -2853,11 +2853,30 @@ public function getDashboardActions()
             ];
         }
 
-        $actions = $this->configureDashboardButtons($actions);
+        $actions = $this->configureDashboardActions($actions);
+
+        // NEXT_MAJOR: Remove this code.
+        $newActions = $this->configureDashboardButtons($actions);
+        if ($newActions !== $actions) {
+            @trigger_error(
+                'configureDashboardButtons() is deprecated since sonata-project/admin-bundle 3.x'
+                .' and will not be used in version 4.0. Use configureDashboardActions() instead.'
+            );
+
+            $actions = $newActions;
+        }
 
         foreach ($this->getExtensions() as $extension) {
             // NEXT_MAJOR: remove method check
+            if (method_exists($extension, 'configureDashboardActions')) {
+                $actions = $extension->configureDashboardActions($this, $actions);
+            }
+            // NEXT_MAJOR: remove this code.
             if (method_exists($extension, 'configureDashboardButtons')) {
+                @trigger_error(
```

Comment:
Wouldn't this trigger an unavoidable deprecation for any extension extending `AbstractAdminExtension`?

Reply:
Fixed